### DISCUSSION TITLE
DOM은 무엇인가요?

### DISCUSSION THREAD

**Discussion by javascriptbiggosoo** [wanted-9th-3team/frontend-study#24] [프론트 기술 질문 공간]
그리고 JavaScript와 어떤 관계인가요?

**Comment by skyhanull** [comment]
DOM은 문서객체모델로 HTML 문서를 계층적 구조와 정보로 표현하며, 이를 제어할 수 있는 프로퍼티와 메서드를 제공하는 트리 자료구조입니다.
즉 DOM은 Javascript가 쉽게 웹 페이지에 접근하여 조작할 수 있게끔 연결시켜주는 역할을 담당합니다.

**Comment by michoball** [comment]
DOM 은 Document Object Model 의 약자로 MDN 문서에 따르면 HTML이나 XML문서를 나타내고 상호작용하는 API 라고 합니다.

HTML 로 작성된 web page가 브라우저 안에서 화면에 나타나고 event에 반응하고 값을 입력받는 등의 기능들을 할 수 있는 객체들로 실체화 된 형태라고 할 수 있습니다. 

DOM 조작은 조작할 수 있는 라이브러리가 있으면 어떤 언어로든 가능하나 웹브라우져가 이해하는 언어가 JS이기 때문에  대표적으로 JS로 조작을 합니다.

**Comment by WooGie911** [comment]
DOM이란 문서 객체 모델( The Document Object Model )의 약자로 
HTML, XML 문서의 프로그래밍 interface 입니다.

웹 페이지는 일종의 문서(document)이며 이 문서는 웹 브라우저를 통해 그 내용이 해석되어 
웹 브라우저 화면에 나타나거나 HTML 소스 자체로 나타나기도 합니다.

이때 DOM은 프로그래밍 언어가 DOM 구조에 접근할 수 있는 방법을 제공하여 
그들이 문서 구조, 스타일, 내용 등을 변경할 수 있게 돕습니다.

또한 DOM은 웹 페이지의 객체 지향 표현이며, 
자바스크립트와 같은 스크립팅 언어를 이용해 DOM을 수정할 수 있습니다.

**Comment by javascriptbiggosoo** [comment]
Document Object Model이란? 웹 페이지를 구성하는 JavaScript object들의 집합입니다.
DOM은 특정 언어에 종속되지 않고 브라우저에서 독립적으로 디자인 되었습니다.
JS로 DOM을 제어할 수 있기 때문에 DOM 자체를 JS의 요소라고 혼동할 수 있는데, 그저 JS와 웹 페이지가 interact 할 수 있도록 해주는 매개체(web API)일 뿐입니다.

**Comment by BaekYuri** [comment]
DOM(Document Object Model)은 웹페이지의 구성 요소들을 구조화 시키고 표현하여 프로그래밍 언어가 해당 문서에 접근하여 조작할 수 있도록 하는 인터페이스 입니다. 트리 자료구조로 구축되기 때문에 DOM Tree라고 부르기도 합니다.
자바스크립트는 DOM을 조작할 수 있는 대표적인 언어입니다. 현재 웹 브라우저에서 DOM을 조작할 수 있는 언어는 자바스크립트 뿐이지만, 다른 프로그래밍 언어로도 DOM을 구현할 수 있습니다.

**Comment by bona373737** [comment]
DOM(Document Object Model)은 브라우저 렌더링 엔진의 HTML parser에 의해 생성된 트리 구조의 Node 객체 모델입니다.
